state/execution.go

```diff
@@ -307,19 +307,20 @@ func (blockExec *BlockExecutor) CheckBlock(block *types.Block) error {
 			return fmt.Errorf("invalid transaction, code: %d", tx.Code)
 		}
 	}
-	resultHash := CheckBlockResponseResultHash(resp)
-	if !bytes.Equal(resultHash, block.Header.LastResultsHash.Bytes()) {
+	resultHash := ABCIResponsesResultsHash(&tmstate.ABCIResponses{
+		DeliverBlock: &abcix.ResponseDeliverBlock{Events: resp.Events, DeliverTxs: resp.DeliverTxs}})
+	if !bytes.Equal(resultHash, block.Header.ResultsHash.Bytes()) {
 		blockExec.logger.Error(
-			"resultHash mismatch. ResultHash in ResponseCheckBlock: %X\n ResultHash in block header: %X",
-			resultHash, block.Header.LastResultsHash,
+			"resultHash mismatch", "resultHash in ResponseCheckBlock", resultHash,
```

Comment:
if you log errors with key/value pair, make sure your key doesn't include white spaces, otherwise I remember the log format will be messed up